sotodlib/io/load_smurf.py

```diff
@@ -2125,20 +2133,32 @@ def _get_timestamps(streams, load_type=None, linearize_timestamps=True):
             if None, uses highest precision version possible. integer values
             will use the TimingParadigm class for indexing
         linearize_timestamps : bool
-          if true and using unix timing, linearize the timing based on the 
+          if true and using unix timing, linearize the timing based on the
           frame counter
     """
     if load_type is None:
         # determine the desired loading type. Expand as logic as
         # data fields develop
         if "primary" in streams:
-            if "UnixTime" in streams["primary"]:
+            if np.diff(io_load.hstack_into(None,
```

Comment:
I'm concerned that this is the only way we know we're using the timing system? Is there not some sort of header information or status information that just says "we're using the timing system"? If it's something that we need to load from the status frame or something then we can add it into `load_file`, but if there's nothing.... what can we add to SMuRF to get us that info?

Also, you're sure this math is backwards compatible? It's not going to trip up data where there isn't a timing system?

Reply:
I think this is the best and easiest way to check for the time being. There is technically the "timing link up" variable, but I don't love relying on a status variable since that could potentially change mid-stream (though you could make the argument that it invalidates the data anyway...). Once we get more data and understand the failure modes better I'm totally fine changing this though.

That being said, the mean is probably not the right thing to use here since there are large negative diffs when the counter resets. `np.mean(np.abs(np.diff(Counter0)))` seems more reliable.